bundlepath.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2015 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package charmrepo
+
+import (
+	"os"
+	"path/filepath"
+
+	"gopkg.in/errgo.v1"
+	"gopkg.in/juju/charm.v6-unstable"
+)
+
+// BundlePath is used to access a bundle at a given path.
+type BundlePath interface {
+	// Bundle returns the bundle represented by this path.
+	Bundle() (charm.Bundle, *charm.URL)
+}
+
+// bundlePath represents a path (eg local directory) containing a
+// single bundle.
+type bundlePath struct {
+	Name       string
+	BundleInfo charm.Bundle
+}
+
+var _ BundlePath = (*bundlePath)(nil)
+
+// NewBundlePath creates and returns an instance used to
+// open a bundle located at the given path.
+func NewBundlePath(path string) (BundlePath, error) {
```

Comment:
Why bother with the BundlePath struct and interface? We could just return (charm.Bundle, *charm.URL, error) from this function and it would be simpler and no less useful, I think.